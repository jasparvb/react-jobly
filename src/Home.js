import React from "react";

function Home({loggedIn}) {
  return (
    <section className="Home col-md-12 mt-5">
      <div class="container text-center">
        <h1 class="mb-4 font-weight-bold">Jobly</h1>
        <p class="lead">All the jobs in one, convenient place.</p>
        {loggedIn ? <h2>Welcome Back!</h2> : <a class="btn btn-primary font-weight-bold" href="/login">Log in</a>}
      </div>
    </section>
  );
}

export default Home;
