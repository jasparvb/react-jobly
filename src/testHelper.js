import React from "react";
import UserContext from "./UserContext";

const demoUser = {
  username: "test",
  first_name: "test",
  last_name: "mctest",
  email: "test@test.net",
  photo_url: null,
  jobs: []
};

const UserProvider = ({ children, user = demoUser }) => (
  <UserContext.Provider value={{user}}>
    {children}
  </UserContext.Provider>
);

export { UserProvider };
