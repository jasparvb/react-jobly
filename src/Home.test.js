import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Home from "./Home";
import { UserProvider } from "./testHelper";

it("renders without crashing", function() {
  render(
    <MemoryRouter>
      <UserProvider>
        <Home />
      </UserProvider>
    </MemoryRouter>
  );
});

it("matches snapshot", function() {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <Home />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when logged out", function() {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider currentUser={null}>
        <Home />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
