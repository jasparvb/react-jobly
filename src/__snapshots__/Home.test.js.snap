// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="Home col-md-12 mt-5"
  >
    <div
      class="container text-center"
    >
      <h1
        class="mb-4 font-weight-bold"
      >
        Jobly
      </h1>
      <p
        class="lead"
      >
        All the jobs in one, convenient place.
      </p>
      <h2>
        Welcome Back!
      </h2>
    </div>
  </section>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <section
    class="Home col-md-12 mt-5"
  >
    <div
      class="container text-center"
    >
      <h1
        class="mb-4 font-weight-bold"
      >
        Jobly
      </h1>
      <p
        class="lead"
      >
        All the jobs in one, convenient place.
      </p>
      <h2>
        Welcome Back!
      </h2>
    </div>
  </section>
</DocumentFragment>
`;
